### PR TITLE
MINOR: Use ListOffsets request instead of SimpleConsumer in LogOffsetTest

### PR DESCRIPTION
**Pull request by ijuma**
Included a few clean-ups related to unused variables in tests.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala [hunk 1]
```diff
@@ -1161,14 +1161,6 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
 
   private def listenerPrefix(name: String): String = new ListenerName(name).configPrefix
 
-  private def configureDynamicKeystoreInZooKeeper(kafkaConfig: KafkaConfig, brokers: Seq[Int], sslProperties: Properties): Unit = {
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala]: @rajinisivaram is it expected that this is unused?

Comment (rajinisivaram) [reply core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala]: @ijuma I had removed the initialization of keystores in the test before starting brokers in an earlier commit, but am using this method in this PR: https://github.com/apache/kafka/pull/4898

**Review by rajinisivaram — APPROVED** [review]
@ijuma Left a comment about the change to DynamicBrokerReconfigurationTest. I can update the other PR if you want to merge this. Apart from that, LGTM

File: core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala]: OK, I will add it back.

**Comment by ijuma** [comment]
@rajinisivaram, rebased and updated.

**Review by rajinisivaram — APPROVED** [review]
Thanks @ijuma, LGTM

**Comment by ijuma** [comment]
One test failure between both Jenkins jobs and not related. Merging to trunk and 2.0.

**Pull request merged**